app/src/main/java/eu/faircode/netguard/Receiver.java

```diff
@@ -224,13 +224,9 @@ public static void upgrade(boolean initialized, Context context) {
                 Log.i(TAG, "Initializing sdk=" + Build.VERSION.SDK_INT);
                 editor.putBoolean("whitelist_wifi", false);
                 editor.putBoolean("whitelist_other", false);
-                if (Build.VERSION.SDK_INT == Build.VERSION_CODES.LOLLIPOP_MR1) {
```

Comment:
No bug there, from my perspective, SDK21 aka LOLLIPOP covers 5.0.x, where you had a lot of complains.
SDK22 aka LOLLIPOP_MR1 covers 5.1.x where you had no complains (except that Note4 :P)

You are covering that Note4 case?